_posts/blog/2016-09-13-code-reviews.md

```diff
@@ -0,0 +1,111 @@
+
+The author will likely feel not so pained by the rule, and decide to follow it. If not, a healthy discussion arises that might change something for the better for the whole team.
+
+As a general rule of thumb: __do not define too many rules and processes__. In a review: start with understanding the requirements, think about potential solutions also from a high-level, architectural and design-perspective. __How would you have solved the problem?__ Then look at the solution. __Find real bugs__. Only beginners focus on cosmetics (and usually miss real flaws).
+
+## Cosmetics are cosmetics
+
+Cosmetics are called cosmetics because they are __not substantial__. Bad formatting, incomplete documentation, bad naming or other trivial things should be handled as what they are: cosmetics. Don't ignore them, __good make-up can make a difference__. But as a reviewer, double-check to make sure your attitude fits the importance of the advice. Nothing is worse than bad vibes because of things that do not have a real impact.
+
+## Don't Override - Contribute
+
+It is generally a good idea for the reviewer to fix small issues during the code review. You can still point the author at the changes, but it doesn't make sense to invest the time twice. If you can fix a spelling error on the fly - why not do it?
+
+But don't change the design completely. Do not override the __substance__ of what your author wanted to do. Help the author, don't control the author.
+
+
+## Don't Only Focus On The Bad Things
+
+Try to emphasize good things about the code under review. One of the developers from my team, Peter Tackage, recently stated: _If I give compliments to someone's implementation then it can lessen the unwanted negative or criticizing aspects of a review. Something like: 'nice solution' or 'neat' when merited, can reinforce a positive mindset in reviews and communicate to the author that the reviewer is truly interested in solving the problem, rather than simply solving the problem their way_ - and I agree.
+Sometimes that's a cultural thing. Germans, like me, for instance tend to focus on problems. Maybe that is even an european issue while people from the USA are well-trained in praising other peoples sweet spots.
+In communication- and management-trainings people learn about the burger-strategy. If you want to criticize someone, wrap it in two nice things. Whereas I resent tactical/manipulative behavior there is a point of truth in this. If you try to see the good, talking about the bad has a completely different taste.
+
+## And The Author...
+
+This text has focused on the reviewer's behavior until now. Of course there is another book to be written about the author and their behavior.
+About how to handle and accept critical feedback; not only about how to give such.
+
+In general you can adopt everything said; try to trust the reviewer, assume positive intent.
+
+Be aware of the fact that written word always strongly underlies interpretation.
```

Comment:
Of course they go for communication in general. It has more impact when receiving critics from the reviewer though. I still think this can stay here. Do you strongly disagree?